### PR TITLE
feat : filter and videos added

### PR DESCRIPTION
**Pull request by ragini07**
1.Added all pages
2.Filter videos by category
3.Navigate to Video Listing page with selected Category from Home Page
4.get videos and categories from backend
5.Single video page route

https://deploy-preview-1--fitsuvideolibrary.netlify.app/

### CODE REVIEW

File: src/Components/Home/Home.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const [category , setCategory] = useState([])
+  const [isLoading , setIsLoading] = useState(false)
+  const navigate = useNavigate()
+  const {filterState , dispatchFilterState} = useVideos()
+  useEffect(() => {
+   ( async () => {
+        try{
+          setIsLoading(true)
+          const {data} = await axios.get('/api/categories')
+          console.log(data.categories)
```
Comment (manismk) [inline src/Components/Home/Home.js]: can remove the console log

File: src/Components/SideBar/SideBar.css [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    z-index: 5;
+    display: none;
+    color :  var(--lg-color);
+}
+
+@media(max-width:800px){
+    /* .main-container{
+        padding: 6rem 0 0 0 ;
+    
+    } */
```
Comment (manismk) [inline src/Components/SideBar/SideBar.css]: remove the commented code if not needed

File: src/Components/VideoDetail/VideoDetail.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+function VideoDetail() {
+    const [video , setVideo] = useState({})
+    const [isLoading, setIsLoading] = useState(false)
+    const {id} = useParams();
+    useEffect(() => {
+        (  async  () => {
+                try{
+                    setIsLoading(true)
+                    const {data} = await axios.get(`/api/video/${id}`)
+                    console.log(data.video)
```
Comment (manismk) [inline src/Components/VideoDetail/VideoDetail.js]: can remove the console log

File: src/Service/services.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import axios from "axios"
+
+export const getVideosFromServer = async (setVideos) =>{
+ try {
+    const {data , status} = await axios.get('/api/videos')
+    if(status === 200) 
+        setVideos(() => data.videos)
+
+    console.log('get data from server', data)
```
Comment (manismk) [inline src/Service/services.js]: can remove the console log

File: src/index.css [hunk 5]
```diff
@@ -1,13 +1,109 @@
...
+  --primary-hover: #e0582b;
+  --secondary-color: rgb(37, 37, 37);
+  --secondary-hover: rgb(49, 49, 49);
+  --font-primary-color: black;
+  --font--secondary-color: rgb(58, 58, 58);
+  --lg-color: white;
+  --body-background-color: #f8f8f8;
+  --light-gray: rgb(184, 182, 182);
+  --success-color : green;
+  --yellow-color : rgb(255, 180, 0);
```
Comment (manismk) [inline src/index.css:17]: Be consistent in providing colors, Either use hex code or RGB value.

**Review by manismk — APPROVED** [review]
Rest looks good. clean UI
Good naming convention.

Suggestion: Try to use barrel export for images

File: src/Components/Header/Header.css [hunk 6]
```diff
@@ -0,0 +1,34 @@
+.main-navhead{
+    display: flex;
+    align-items: center;
+    height: 60px;
```
Comment (znjs) [inline src/Components/Header/Header.css:4]: Use relative units for responsiveness

File: src/Components/Home/Home.css [hunk 7]
```diff
@@ -0,0 +1,89 @@
+
+/* showcase */
+.showcase{
+    /* width: 100%; */
```
Comment (znjs) [inline src/Components/Home/Home.css]: Can remove the commented code

File: src/Components/Home/Home.css [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    height: 400px;
```
Comment (znjs) [inline src/Components/Home/Home.css:4]: Use relative units

File: src/Components/Home/Home.css [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    margin: 2rem auto;
+    display: grid;
+    grid-template-columns: repeat(2,1fr);
+    grid-gap:1.5rem;
+    max-width: 58rem;
+
+}
+.home-card{
+    position: relative;
+    /* max-width: 200px; */
```
Comment (znjs) [inline src/Components/Home/Home.css:31]: Can remove commented code

File: src/Components/SideBar/SideBar.css [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    flex-direction: column;
+    align-items: center;
+    width: 20%;
+    height: 100%;
+    background-color : var(--primary-color);
+    position: fixed;
+    left: 0;
+    z-index: 3;
+    border-top: 1px solid var(--lg-color);
+    /* border-right: 1px solid var(--primary-color); */
```
Comment (znjs) [inline src/Components/SideBar/SideBar.css]: Can remove commented code if possible

File: src/Context/data-context.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useReducer, useState } from "react";
+import { filterReducerFtn } from "../Reducer/filterReducer";
+import { initialFilterState } from "../Utils/constants";
+
+const VideosContext = createContext();
+
+const  VideosProvider= ({ children }) => {
+
+    const [videos , setVideos ] = useState([])
+    const [categories , setCategories] = useState([])
```
Comment (znjs) [inline src/Context/data-context.js:10]: Can include these inside reducer and remove `useState`

File: src/Reducer/filterReducer.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { initialFilterState } from "../Utils/constants";
+
+export const filterReducerFtn = (state , action) => {
+    switch(action.type) {
+        case "CATEGORY" :
```
Comment (znjs) [inline src/Reducer/filterReducer.js:5]: Use constants to avoid string typo bugs

File: src/Reducer/filterReducer.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+            return {...state , category : action.payload }       
+        case "CLEAR_FILTER" :
```
Comment (znjs) [inline src/Reducer/filterReducer.js:7]: Use constants to avoid string typo bugs

File: src/Service/services.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+ }catch(error){
+     console.log(error)
```
Comment (znjs) [inline src/Service/services.js]: if possible use `console.error` to give a semantic meaning

File: src/Service/services.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+ }
+} 
+export const getCategoryFromServer = async (setCategories) =>{
+ try {
+    const {data , status} = await axios.get('/api/categories')
+    if(status === 200) 
+        setCategories(() => data.categories)
+    console.log('get data from server', data)
+ }catch(error){
+     console.log(error)
```
Comment (znjs) [inline src/Service/services.js]: if possible use console.error to give a semantic meaning

File: src/index.css [hunk 16]
```diff
@@ -1,13 +1,109 @@
...
+.searchbar{
+  display: block;
+}
+.link-btn{
+  color : var(--lg-color)
+}
+.main-container{
+  display: flex;
+  /* gap:4rem; */
+  /* padding: 5rem; */
```
Comment (znjs) [inline src/index.css]: Can remove the commented code

**Review by znjs — APPROVED** [review]
Good naming conventions used
good folder structure
Liked that you already made a not found page

File: src/Components/Home/Home.js (hunk 1 above)
Comment (ragini07) [reply src/Components/Home/Home.js]: done

File: src/Components/VideoDetail/VideoDetail.js (hunk 3 above)
Comment (ragini07) [reply src/Components/VideoDetail/VideoDetail.js]: done

File: src/index.css (hunk 5 above)
Comment (ragini07) [reply src/index.css:17]: I will use RGB value for all at the end.. As I am not liking the color combination I used in UI I will change it a bit and fix this part as well.

File: src/Components/Header/Header.css (hunk 6 above)
Comment (ragini07) [reply src/Components/Header/Header.css:4]: Will fix all issues in responsiveness after I am done with functionality

File: src/Components/Home/Home.css (hunk 9 above)
Comment (ragini07) [reply src/Components/Home/Home.css:31]: I may use this for small screen , so for now kept it commented.If not required I will remove it later

File: src/Reducer/filterReducer.js (hunk 12 above)
Comment (ragini07) [reply src/Reducer/filterReducer.js:5]: yes Thankyou for this suggestion.Will use constants from now on

**Pull request merged**